pkg/controller/machine.go

```diff
@@ -507,17 +507,17 @@ func (c *Controller) ensureInstanceExistsForMachine(prov cloud.Provider, machine
 				return fmt.Errorf("failed get userdata: %v", err)
 			}
 
-			// We will create the instance, so ensure finalizer is there
-			machine, err = c.ensureDeleteFinalizerExists(machine)
-			if err != nil {
-				return err
-			}
 			// Create the instance
 			if providerInstance, err = c.createProviderInstance(prov, machine, userdata); err != nil {
 				c.recorder.Eventf(machine, corev1.EventTypeWarning, "CreateInstanceFailed", "Instance creation failed: %v", err)
 				message := fmt.Sprintf("%v. Unable to create a machine.", err)
 				return c.updateMachineErrorIfTerminalError(machine, machinev1alpha1.CreateMachineError, message, err, "failed to create machine at cloudprover")
 			}
+			// We created the instance, so ensure finalizer is there
+			machine, err = c.ensureDeleteFinalizerExists(machine)
```

Comment:
Days since last moving `ensureDeleteFinalizerExists` some other place: ~14~ -> 0
(Not really checked when we changed it last time)